MP Health test development

istraka opened the pull request:
https://eap-qe-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/eap-7.x-microprofile-testsuite/40/
https://eap-qe-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/eap-7.x-microprofile-testsuite/42/

Please make sure your PR meets the following requirements:
- [x] Pull Request contains a description of the changes
- [x] Pull Request does not include fixes for multiple issues/topics
- [x] Code is formatted, imports ordered, code compiles and tests are passing
- [x] Link to the passing job is provided
- [x] Code is self-descriptive and/or documented
- [x] Description of the tests scenarios is included (see #46)

honza-kasik commented on microprofile-health/src/main/java/org/jboss/eap/qe/microprofile/health/integration/CustomConfigSource.java:
```diff
@@ -0,0 +1,43 @@
+
+import org.eclipse.microprofile.config.spi.ConfigSource;
+
+public class CustomConfigSource implements ConfigSource {
```
A brief JavaDoc explaining what this is good for would be great!

honza-kasik commented on microprofile-health/src/main/java/org/jboss/eap/qe/microprofile/health/integration/CustomConfigSource.java:
```diff
@@ -0,0 +1,43 @@
+import org.eclipse.microprofile.config.spi.ConfigSource;
+
+public class CustomConfigSource implements ConfigSource {
+    public static final String FILEPATH_PROPERTY = "config.source.properties.path";
```
Wouldn't be `PROPERTIES_FILE_PATH` better? WDYT?

honza-kasik commented on microprofile-health/src/main/java/org/jboss/eap/qe/microprofile/health/integration/CustomConfigSource.java:
```diff
@@ -0,0 +1,43 @@
+                props.put(key, properties.getProperty(key));
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
```
Do you really want to just print the stack trace and continue?

istraka replied:
Good point. I'll throw an `RuntimeException`

honza-kasik commented on microprofile-health/src/main/java/org/jboss/eap/qe/microprofile/health/integration/CustomConfigSourceProvider.java:
```diff
@@ -0,0 +1,13 @@
+import org.eclipse.microprofile.config.spi.ConfigSource;
+import org.eclipse.microprofile.config.spi.ConfigSourceProvider;
+
+public class CustomConfigSourceProvider implements ConfigSourceProvider {
```
Just a brief JavaDoc would be IMHO much better.

honza-kasik commented on microprofile-health/src/test/java/org/jboss/eap/qe/microprofile/health/integration/FailSafeCDICustomConfigSourceHealthTest.java:
```diff
@@ -0,0 +1,92 @@
+
+        @Override
+        public void tearDown(ManagementClient managementClient, String s) throws Exception {
+            OnlineManagementClient client = ManagementClientProvider.onlineStandalone(managementClient);
```
Please close client.

istraka replied:
Actually, the client can not be closed. Doing so would also close `managementClient` managed by arquillian which is something we do not want to. I would not be able to deploy then since  `managementClient` would be closed.

honza-kasik replied:
Thanks - I understand limitations which you are running into but there is still an open client in the end...

istraka replied:
I see. There are two options:
- `ManagementClientProvider.onlineStandalone()` - this always opens a new connection which can be closed
- `ManagementClientProvider.onlineStandalone(managementClient)` - this just wraps `managementClient` and no other resources are opened thus it can be left opened. Since `managementClient` lifecycle is managed by Arquillian It must be left opened. And is closed by Arquillian at the end.

honza-kasik replied:
Thanks - this issue was discussed off GitHub.

istraka replied:
And I know it is dangerous since Creaper wrapper can change in the future and open a resource which would be left in (2) option. However it is unlikely and @mnovak1 is ok with this (see https://github.com/jboss-eap-qe/eap-microprofile-test-suite/pull/142)

honza-kasik commented on microprofile-health/src/test/java/org/jboss/eap/qe/microprofile/health/integration/FailSafeCDICustomConfigSourceHealthTest.java:
```diff
@@ -0,0 +1,92 @@
+
+        @Override
+        public void setup(ManagementClient managementClient, String s) throws Exception {
+            OnlineManagementClient client = ManagementClientProvider.onlineStandalone(managementClient);
```
Please close client.

istraka replied:
Actually, the client can not be closed. Doing so would also close `managementClient` managed by arquillian which is something we do not want to. I would not be able to deploy then since  `managementClient` would be closed.

honza-kasik replied:
Thanks - I understand limitations which you are running into but there is still an open client in the end...

mnovak1 replied:
I was looking at the code of Arquillian extension and it looks like that `ManagementClient` is simply injected there as in ordinary tests where we don't close it as well. However I'm not sure what is the object lifecycle for `@Inject`. Still I don't see as issue.

honza-kasik commented on microprofile-health/src/test/java/org/jboss/eap/qe/microprofile/health/integration/FailSafeCDICustomConfigSourceHealthTest.java:
```diff
@@ -0,0 +1,92 @@
+@RunWith(Arquillian.class)
+@RunAsClient
+@ServerSetup({ MicroProfileFTSetupTask.class, FailSafeCDICustomConfigSourceHealthTest.SetupTask.class })
+public class FailSafeCDICustomConfigSourceHealthTest extends FailSafeCDIHealthDynamicBaseTest {
```
JavaDoc would be nice here

mnovak1 replied:
I would like to have see what the test is testing in JavaDoc

istraka replied:
It is in `FailSafeCDIHealthDynamicBaseTest`.  `FailSafeCDICustomConfigSourceHealthTest` is only about providing MP Config properties in specific way.

mnovak1 replied:
Ahh, got it. Thanks

honza-kasik commented on microprofile-health/src/test/java/org/jboss/eap/qe/microprofile/health/integration/FailSafeCDICustomConfigSourceHealthTest.java:
```diff
@@ -0,0 +1,92 @@
+
+    @Deployment(testable = false)
+    public static WebArchive createDeployment() {
+        WebArchive webArchive = ShrinkWrap
```
You can return it directly, no need for variable.

honza-kasik commented on microprofile-health/src/test/java/org/jboss/eap/qe/microprofile/health/integration/FailSafeCDICustomConfigSourceProviderHealthTest.java:
```diff
@@ -0,0 +1,97 @@
+
+    @Deployment(testable = false)
+    public static WebArchive createDeployment() {
+        WebArchive webArchive = ShrinkWrap
```
No need for variable

honza-kasik commented on microprofile-health/src/test/java/org/jboss/eap/qe/microprofile/health/integration/FailSafeCDICustomConfigSourceProviderHealthTest.java:
```diff
@@ -0,0 +1,97 @@
+
+        @Override
+        public void setup(ManagementClient managementClient, String s) throws Exception {
+            OnlineManagementClient client = ManagementClientProvider.onlineStandalone(managementClient);
```
Please close client

istraka replied:
Actually, the client can not be closed. Doing so would also close `managementClient` managed by arquillian which is something we do not want to. I would not be able to deploy then since  `managementClient` would be closed.

honza-kasik replied:
Thanks - I understand limitations which you are running into but there is still an open client in the end...

honza-kasik commented on microprofile-health/src/test/java/org/jboss/eap/qe/microprofile/health/integration/FailSafeCDICustomConfigSourceProviderHealthTest.java:
```diff
@@ -0,0 +1,97 @@
+@RunWith(Arquillian.class)
+@RunAsClient
+@ServerSetup({ MicroProfileFTSetupTask.class, FailSafeCDICustomConfigSourceProviderHealthTest.SetupTask.class })
+public class FailSafeCDICustomConfigSourceProviderHealthTest extends FailSafeCDIHealthDynamicBaseTest {
```
Please add a JavaDoc. I have very little idea which are the differences between all these similar looking tests.

honza-kasik commented on microprofile-health/src/test/java/org/jboss/eap/qe/microprofile/health/integration/FailSafeCDIHealthBaseTest.java:
```diff
@@ -44,6 +46,8 @@ public void before() throws ConfigurationException, InterruptedException, Timeou
                 + "/metrics";
         metricsRequest = given().baseUri(url).accept(ContentType.JSON);
+        // reset MP Config properties
+        new Administration(ManagementClientProvider.onlineStandalone(managementClient)).reload();
```
Client not closed.

istraka replied:
Actually, the client can not be closed. Doing so would also close `managementClient` managed by arquillian which is something we do not want to. I would not be able to deploy then since  `managementClient` would be closed.

honza-kasik commented on microprofile-health/src/test/java/org/jboss/eap/qe/microprofile/health/integration/FailSafeCDIHealthDynamicBaseTest.java:
```diff
@@ -0,0 +1,237 @@
+
+import io.restassured.http.ContentType;
+
+public abstract class FailSafeCDIHealthDynamicBaseTest extends FailSafeCDIHealthBaseTest {
```
Please add a brief JavaDoc.

honza-kasik commented on microprofile-health/src/test/resources/org/jboss/eap/qe/microprofile/health/integration/health.properties:
```diff
@@ -0,0 +1,4 @@
+dummy.live=1
```
Can you please add a comment to this file explaining its purpose?

honza-kasik commented on microprofile-health/src/test/resources/org/jboss/eap/qe/microprofile/health/integration/configSourceModule.xml:
```diff
@@ -0,0 +1,34 @@
+  ~ 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+  -->
+
+<module xmlns="urn:jboss:module:1.5" name="test.custom-config-source">
```
Can you please add a comment to these files explaining their purpose?

istraka commented:
@honza-kasik thanks for the review. I have addressed issues except the client one. 
https://eap-qe-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/eap-7.x-microprofile-testsuite/41/

honza-kasik commented:
@istraka LGTM, Thanks!

mnovak1 commented on microprofile-health/src/test/java/org/jboss/eap/qe/microprofile/health/integration/FailSafeCDIHealthDynamicBaseTest.java:
```diff
@@ -0,0 +1,242 @@
+     * @tpSince EAP 7.4.0.CD19
+     */
+    @Test
+    @RunAsClient
```
Optional: If everything is `@RunAsClient` then just the class could be annotated and not every test.

istraka commented:
Fixed

mnovak1 commented:
LGTM
